cl/donate/models.py

```diff
@@ -164,3 +166,85 @@ def __str__(self) -> str:
             self.monthly_donation_amount,
             self.get_payment_provider_display(),
         )
+
+
+class NeonWebhookEvents(models.Model):
+    MEMBERSHIP_CREATION = 1
+    MEMBERSHIP_EDIT = 2
+    MEMBERSHIP_DELETE = 3
+    MEMBERSHIP_UPDATE = 4
+    TYPES = (
+        (MEMBERSHIP_CREATION, "createMembership"),
+        (MEMBERSHIP_EDIT, "editMembership"),
+        (MEMBERSHIP_DELETE, "deleteMembership"),
+        (MEMBERSHIP_UPDATE, "updateMembership"),
+    )
+    trigger = models.PositiveSmallIntegerField(
+        help_text="The current membership tier of a user within Neon CRM",
+        choices=TYPES,
+        null=True,
+    )
+    account_id = models.CharField(
+        help_text="Unique identifier assigned by Neon CRM to a customer record",
+        default="",
+        blank=True,
+    )
+    membership_id = models.CharField(
+        help_text="Unique identifier assigned by Neon CRM to a membership record",
+        default="",
+        blank=True,
+    )
+    content = models.JSONField(  # type: ignore
+        help_text="The content of the payload of the POST request.",
+        blank=True,
+        null=True,
+    )
+
+
+@pghistory.track(AfterUpdateOrDeleteSnapshot())
+class NeonMembership(models.Model):
+    BASIC = 1
+    LEGACY = 2
+    TIER_1 = 3
+    TIER_2 = 4
+    TIER_3 = 5
+    TIER_4 = 6
+    TIER_5 = 7
+    PLATINUM = 8
+    TYPES = (
+        (BASIC, "CL Membership - Basic"),
+        (LEGACY, "CL Legacy Membership"),
+        (TIER_1, "CL Membership - Tier 1"),
+        (TIER_2, "CL Membership - Tier 2"),
+        (TIER_3, "CL Membership - Tier 3"),
+        (TIER_4, "CL Membership - Tier 4"),
+        (TIER_5, "CL Membership - Tier 5"),
+        (PLATINUM, "CL Platinum Membership"),
+    )
+    INVERTED = invert_choices_group_lookup(TYPES)
+    user = models.OneToOneField(
+        User,
+        related_name="membership",
+        verbose_name="the user linked to the membership",
+        on_delete=models.CASCADE,
+        unique=True,
+    )
+    neon_id = models.CharField(
+        help_text="Unique identifier assigned by Neon CRM to a membership record",
+        default="",
+        blank=True,
+    )
+    level = models.PositiveSmallIntegerField(
+        help_text="The current membership tier of a user within Neon CRM",
+        choices=TYPES,
+        null=True,
```

Comment:
A bit surprising that this can be null?

Reply:
I'm sorry, this field should not be null.